### PR TITLE
Add unittest(cpu-only) and typecheck for Copy

### PR DESCRIPTION
**Pull request by delta2323**
* Add type check for `Copy` Function 
* Add unittest(cpu-only) for `Copy` Function to test type check.

Related to #123

### CODE REVIEW

File: chainer/functions/copy.py [hunk 1]
```diff
@@ -9,6 +12,20 @@ class Copy(function.Function):
     def __init__(self, out_device):
         self.out_device = out_device
 
+    def check_type_forward(self, in_types):
+        type_check.expect(
+            in_types.size() == 1,
+            in_types[0].dtype == numpy.float32
```
Comment (unnonouno) [inline chainer/functions/copy.py]: Does it need to restrict input type?

File: tests/functions_tests/test_copy.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    def setUp(self):
+        self.x_data = numpy.random.uniform(
+            -1, 1, (10, 5)).astype(numpy.float32)
+        self.gy = numpy.random.uniform(-1, 1, (10, 5)).astype(numpy.float32)
+
+    def test_check_forward_cpu(self):
+        x = chainer.Variable(self.x_data)
+        y = functions.copy(x, -1)
+        gradient_check.assert_allclose(self.x_data, y.data, atol=0, rtol=0)
```
Comment (unnonouno) [inline tests/functions_tests/test_copy.py]: I think values are exactly the same?

Comment (delta2323) [inline tests/functions_tests/test_copy.py]: atol and troll is 0, so I think we do not admit any torelance.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**